src/hex.rs

```diff
@@ -97,21 +97,21 @@ impl<'a> Iterator for HexIterator<'a> {
     type Item = Result<u8, Error>;
 
     fn next(&mut self) -> Option<Result<u8, Error>> {
-        let hi = self.chars.next()?;
-        let lo = self.chars.next().unwrap();
+        let hi = self.iter.next()?;
+        let lo = self.iter.next().unwrap();
         Some(chars_to_hex(hi, lo))
     }
 
     fn size_hint(&self) -> (usize, Option<usize>) {
-        let len = self.chars.as_str().len() / 2;
-        (len, Some(len))
+        let (min, max) = self.iter.size_hint();
```

Comment:
Could this be the culprit of the performance hit?

It uses `slice::Iter`'s size_hint, which uses the following macro, not sure what to make of it:
```rust

// To get rid of some bounds checks (see `position`), we compute the length in a somewhat
// unexpected way. (Tested by `codegen/slice-position-bounds-check`.)
macro_rules! len {
    ($self: ident) => {{
        let start = $self.ptr;
        let diff = ($self.end as usize).wrapping_sub(start as usize);
        let size = mem::size_of(start);
        if size == 0 {
            diff
        } else {
            // Using division instead of `offset_from` helps LLVM remove bounds checks
            diff / size
        }
    }}
}
```

Reply:
Hmm, I don't think so.